### DISCUSSION TITLE
What is "ConstStateDecorator::getFPValues is not supposed to be called " error

### DISCUSSION THREAD

**Discussion by NitishNY** [UPPAALModelChecker/UPPAAL-Meta#146] [Q&A]
I am implementing a template that looks like this:
![image](https://user-images.githubusercontent.com/113758247/190844956-270d7808-4f36-42af-88ac-4d08508d80a0.png)

Upon executing the fetchaction2(num_countermeasures,x) i am getting this error. The code for the function is as follows:

//Defined in declarations
    

    int comp_cost[5]= {3000,1500,2000,4600,8000}; //Cost of SBS component
    int rc_cost[6]= {700,650,1200,420,1000,1400}; // cost of countermeasures
    double ctr[6][5]= {{0.8,0.7,0.4,0.2,0.3},{0.6,0.7,0.4,0.2,0.35},{0.4,0.7,0.8,0.75,0.3},{0.2,0.3,0.75,0.6,0.8},{0.1,0.1,0.2,0.8,0.9}, 
    {0.5,0.5,0.6,0.7,0.5}}; //Effectiveness of countermeasures when deployed for a component of SBS
    int num=6;

//Defined in the template shown above

    id_t fetchaction2(int num, id_t comp_num)
    {
    double maxx=-5000.0;
    id_t index=0;   // defined typedef int[0,9] id_t; in the declaration.
    int i=0;
    for(i=0; i<num; i++) {
        double benefit=(ctr[i][comp_num]*comp_cost[comp_num])-rc_cost[i]; //Benefit Calculation equation
        double temp=(benefit/rc_cost[i]);
        if(maxx<temp) {
            maxx=temp;
            index=i;
        }
    }
    return index;
    }

**Comment by mikucionisaau** [comment]
It has to do with accessing floating point variable values.
In symbolic analysis context, FP variable values are not accessible, so my guess is that you are seeing this either in a symbolic simulator or symbolic query.

On the other hand, you are not supposed to see such message, so this is very likely a bug.
I need more context about what you are doing. I think it should be possible to reproduce the issue with just a single `double` variable, perhaps being declared/accessed inside a function.
